src/common/accountParser.ts

```diff
@@ -180,39 +181,67 @@ const TokenAccountLayout = BL.struct([
   pubkeyField("closeAuthority")
 ])
 
-export const parseTokenAccount = (account: Buffer, accountPubkey: PublicKey) => {
-  const data = TokenAccountLayout.decode(account)
+/** Buffer layout for de/serializing a mint */
+export const RawMintLayout = BL.struct<RawMint>([
+  BL.u32("mintAuthorityOption"),
+  BLU.publicKey("mintAuthority"),
+  u64Field("supply"),
+  BL.u8("decimals"),
+  BLU.bool("isInitialized"),
+  BL.u32("freezeAuthorityOption"),
+  BLU.publicKey("freezeAuthority")
+])
+
+/**
+ * Decode a token account
+ * @param {Buffer} account
+ * @param {PublicKey} accountPubkey
+ * @returns
+ */
+export const parseTokenAccount = (account: Buffer, accountAddress: PublicKey): TokenAccountInfo => {
+  const data = RawTokenAccountLayout.decode(account)
 
   // PublicKeys and BNs are currently Uint8 arrays and
   // booleans are really Uint8s. Convert them
   const decoded: TokenAccountInfo = {
-    address: accountPubkey,
+    address: accountAddress,
     mint: new PublicKey(data.mint),
     owner: new PublicKey(data.owner),
     amount: new BN(data.amount, undefined, "le"),
-    delegate: (data as any).delegateOption ? new PublicKey(data.delegate) : null,
+    delegate: new PublicKey(data.delegate),
     delegatedAmount: new BN(data.delegatedAmount, undefined, "le"),
     isInitialized: (data as any).state != 0,
     isFrozen: (data as any).state == 2,
     isNative: !!(data as any).isNativeOption,
     rentExemptReserve: new BN(0, undefined, "le"), //  Todo: calculate. I believe this is lamports minus rent for wrapped sol
```

Comment:
```suggestion
    delegatedAmount: data.delegatedAmount,
    isInitialized: data.state !== 0,
    isFrozen: data.state === 2,
    isNative: data.isNativeOption === 1,
    rentExemptReserve: data.isNativeOption === 1 ? data.isNative : new BN(0),
```

Rent exempt reserve works like this (from the spl program), which made me write it the way I did above

```
pub struct Account {
    /// If is_native.is_some, this is a native token, and the value logs the rent-exempt reserve. An
    /// Account is required to be rent-exempt, so the value is used by the Processor to ensure that
    /// wrapped SOL accounts do not drop below this threshold.
    pub is_native: COption<u64>
}
```